src/Framework/AspNetCoreAnalyzers/src/Analyzers/Http/HeaderDictionaryIndexerAnalyzer.cs

```diff
@@ -0,0 +1,185 @@
+        ["Content-Range"] = "ContentRange",
+        ["Content-Security-Policy"] = "ContentSecurityPolicy",
+        ["Content-Security-Policy-Report-Only"] = "ContentSecurityPolicyReportOnly",
+        ["Content-Type"] = "ContentType",
+        ["Correlation-Context"] = "CorrelationContext",
+        ["Cookie"] = "Cookie",
+        ["Date"] = "Date",
+        ["ETag"] = "ETag",
+        ["Expires"] = "Expires",
+        ["Expect"] = "Expect",
+        ["From"] = "From",
+        ["Grpc-Accept-Encoding"] = "GrpcAcceptEncoding",
+        ["Grpc-Encoding"] = "GrpcEncoding",
+        ["Grpc-Message"] = "GrpcMessage",
+        ["Grpc-Status"] = "GrpcStatus",
+        ["Grpc-Timeout"] = "GrpcTimeout",
+        ["Host"] = "Host",
+        ["Keep-Alive"] = "KeepAlive",
+        ["If-Match"] = "IfMatch",
+        ["If-Modified-Since"] = "IfModifiedSince",
+        ["If-None-Match"] = "IfNoneMatch",
+        ["If-Range"] = "IfRange",
+        ["If-Unmodified-Since"] = "IfUnmodifiedSince",
+        ["Last-Modified"] = "LastModified",
+        ["Link"] = "Link",
+        ["Location"] = "Location",
+        ["Max-Forwards"] = "MaxForwards",
+        ["Origin"] = "Origin",
+        ["Pragma"] = "Pragma",
+        ["Proxy-Authenticate"] = "ProxyAuthenticate",
+        ["Proxy-Authorization"] = "ProxyAuthorization",
+        ["Proxy-Connection"] = "ProxyConnection",
+        ["Range"] = "Range",
+        ["Referer"] = "Referer",
+        ["Retry-After"] = "RetryAfter",
+        ["Request-Id"] = "RequestId",
+        ["Sec-WebSocket-Accept"] = "SecWebSocketAccept",
+        ["Sec-WebSocket-Key"] = "SecWebSocketKey",
+        ["Sec-WebSocket-Protocol"] = "SecWebSocketProtocol",
+        ["Sec-WebSocket-Version"] = "SecWebSocketVersion",
+        ["Sec-WebSocket-Extensions"] = "SecWebSocketExtensions",
+        ["Server"] = "Server",
+        ["Set-Cookie"] = "SetCookie",
+        ["Strict-Transport-Security"] = "StrictTransportSecurity",
+        ["TE"] = "TE",
+        ["Trailer"] = "Trailer",
+        ["Transfer-Encoding"] = "TransferEncoding",
+        ["Translate"] = "Translate",
+        ["traceparent"] = "TraceParent",
+        ["tracestate"] = "TraceState",
+        ["Upgrade"] = "Upgrade",
+        ["Upgrade-Insecure-Requests"] = "UpgradeInsecureRequests",
+        ["User-Agent"] = "UserAgent",
+        ["Vary"] = "Vary",
+        ["Via"] = "Via",
+        ["Warning"] = "Warning",
+        ["WWW-Authenticate"] = "WWWAuthenticate",
+        ["X-Content-Type-Options"] = "XContentTypeOptions",
+        ["X-Frame-Options"] = "XFrameOptions",
+        ["X-Powered-By"] = "XPoweredBy",
+        ["X-Requested-With"] = "XRequestedWith",
+        ["X-UA-Compatible"] = "XUACompatible",
+        ["X-XSS-Protection"] = "XXSSProtection",
+    };
+
+    private static void AddDiagnosticWarning(OperationAnalysisContext context, Location location, string headerName, string propertyName)
+    {
+        var properties = new Dictionary<string, string>
```

Comment:
Nit: Use an `ImmutableDictionaryBuilder` here.

I think it's better to leverage that for property bags in our analyzers instead of initializing an intermediate `Dictionary` everytime.